src/odemis/gui/cont/acquisition.py

```diff
@@ -693,7 +693,7 @@ def _on_streams_change(self, streams):
         """
         # removing a stream 
         # for every entry in the list, is it also present in the .streams?
-        for i in range(self._panel.streams_chk_list.GetCount() - 1, -1, -1):
+        for i in range(self._panel.streams_chk_list.GetCount() - 1):
```

Comment:
No, the reverse was doing something. It ensures that the index you have and the current list index match.
For example If you have a list with 5 streams [A, B, C, D, E], and stream C and D disappeared, when you iterate in normal order, you first delete item 2 (OK, the list becomes [A, B, D, E]), and then item 3... but item 3 is E!

If you delete in reverse order, you don't have this issue.
=> Put back the reverse order, and add a comment explaining this.
Note that typically, when the streams are removed by the user, only one stream is removed at a time, so such error is not triggered. This will happen when the streams are deleted by some code, for instance when changing project.